pkg/api/register.go

```diff
@@ -118,52 +118,67 @@ func AddToScheme(scheme *runtime.Scheme) {
 func (obj *Pod) GetObjectMeta() meta.Object                                  { return &obj.ObjectMeta }
 func (obj *Pod) GetObjectKind() unversioned.ObjectKind                       { return &obj.TypeMeta }
 func (obj *PodList) GetObjectKind() unversioned.ObjectKind                   { return &obj.TypeMeta }
+func (obj *PodList) GetListMeta() meta.List                                  { return &obj.ListMeta }
```

Comment:
If we can remove that boilerplate that's good.

On Jun 7, 2016, at 4:49 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/api/register.go
<https://github.com/kubernetes/kubernetes/pull/26964#discussion_r66149879>:

> @@ -118,52 +118,67 @@ func AddToScheme(scheme *runtime.Scheme) {
>  func (obj *Pod) GetObjectMeta() meta.Object                                  { return &obj.ObjectMeta }
>  func (obj *Pod) GetObjectKind() unversioned.ObjectKind                       { return &obj.TypeMeta }
>  func (obj *PodList) GetObjectKind() unversioned.ObjectKind                   { return &obj.TypeMeta }
> +func (obj *PodList) GetListMeta() meta.List                                  { return &obj.ListMeta }

Really? https://play.golang.org/p/kwljrT6nSk
… <#>
On Tue, Jun 7, 2016 at 11:37 AM, Clayton Coleman ***@***.***> wrote: In
pkg/api/register.go <#26964 (comment)
<https://github.com/kubernetes/kubernetes/pull/26964#discussion_r66128145>>
: > @@ -118,52 +118,67 @@ func AddToScheme(scheme *runtime.Scheme) { > func
(obj *Pod) GetObjectMeta() meta.Object { return &obj.ObjectMeta } > func
(obj *Pod) GetObjectKind() unversioned.ObjectKind { return &obj.TypeMeta }
> func (obj *PodList) GetObjectKind() unversioned.ObjectKind { return
&obj.TypeMeta } > +func (obj *PodList) GetListMeta() meta.List { return
&obj.ListMeta } It has to be a pointer so it can mutate - if you
implemented on a TypeMeta it would not have access to a pointer to type
meta and your Set* calls would silently be lost. — You are receiving this
because you are on a team that was mentioned. Reply to this email directly,
view it on GitHub <
https://github.com/kubernetes/kubernetes/pull/26964/files/6b0110a098af230d6fd5c3ad2b171ab4730c50dc#r66128145>,
or mute the thread <
https://github.com/notifications/unsubscribe/AAnglq-XScfYO7oxMD3a0a_47ae2Pmqbks5qJbqGgaJpZM4Iv_PB>
.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/26964/files/6b0110a098af230d6fd5c3ad2b171ab4730c50dc#r66149879>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p3KHb0N2vlNaBZpVL_SSHmYbHHNcks5qJdlagaJpZM4Iv_PB>
.